src/uct/tcp/tcp_iface.c

```diff
@@ -766,7 +772,10 @@ ucs_status_t uct_tcp_query_devices(uct_md_h md,
             continue;
         }
 
-        if (!ucs_netif_is_active(entry->d_name)) {
+        uct_md_iface_config_read(md, "tcp", NULL, NULL, &config);
+        iface_config = ucs_derived_of(config, uct_tcp_iface_config_t);
+
+        if (!ucs_netif_is_active(entry->d_name, iface_config->loopback_enable)) {
```

Comment:
maybe it worth to introduce `ucs_netif_is_loopback` which gets iface flags and checks `(ifr.ifr_flags & IFF_LOOPBACK)`?
so it will be like this:
```suggestion
        if (!ucs_netif_is_active(entry->d_name) ||
            (iface_config->loopback_enable && ucs_netif_is_loopback(entry->d_name))) {
```
wdyt?